OSDOCS-7495: add backup and restore (#63608)
openshift/openshift-docs: modules/microshift-backing-up-manually.adoc

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-backing-up-manually_{context}"]
= Backing up {product-title} data manually

> If you are not using an OSTree system, you can back up data manually at any time. Data backups are advised prior to any system updates. Backups are created in the `/var/lib/microshift-backups` directory by default. You can also specify a directory by running an optional backup command.

Reviewer: > at any time

Sort of, with a caveat: `microshift.service` needs to be stopped before attempting to backup (`# systemctl stop microshift.service`).
But I see it's included in Prerequisites below, so I don't know if we need to be more precise in this line

> You can also specify a directory by running an optional backup command.

(nit on _command_ vs _option_)
From my technical POV which I'm not sure how translate to customer doc but i'll mention it:
- You always need to run a backup command (this is the `backup` in `microshift backup`)
- To specify different storage dir (parent dir of backup dir), you use option `--storage`.

.Prerequisites
* You have root access to the host.

> * {product-title} and related workloads are stopped.

Reviewer: What's under "related workloads"?
Technically we don't require user to stop Pods (workloads) before backing up

Reviewer 2: I think this is referring to the volume snapshot creation phase of the backup.  It's definitely recommended that users's stop their stateful workloads before creating the snapshot.

Author: Yes, I was thinking about that as a best practice: Stop everything before you backup anything.

Reviewer 3: We want admin processes to disrupt workloads as little as possible. Since the `microshift backup` command does not back up user workloads or take snapshots, the user workload pods can stay online.

.Procedure
. Manually create a backup file using the default parent directory, `/var/lib/microshift-backups`, by running the following command:
+
[source, terminal]
----

> # microshift backup --storage

Reviewer: `--storage` allows to use different parent dir than the default.
Using the opt, the command would look like
```
# microshift backup --storage /var/lib/custom-storage-location
```

To just create a backup, with default name and storage, it's:
```
# microshift backup
```

----

> . Optional. Manually create a backup file specifying the parent directory in the <name> value by running the following command:

Reviewer: `--name` is used to specify name of the backup, not the parent directory (that's the `--storage`)

+
[source, terminal]
----

> # microshift backup --<name>

Reviewer: ```suggestion
# microshift backup --name <name>
```

Author: @pmtk Can the admin use both a custom name and storage location in the same command?

----
+
.Verification

> . You can verify that the backup was created by viewing the file in the specified directory.

Reviewer: The backup is not a file, it's a directory; it's `<name>` directory in the `<storage>` directory.
For example: `/var/lib/microshift-backups/my-custom-backup/` or `/var/lib/custom-storage-location/another-custom-backup/`

So the schema is: `<storage>/<name>`

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-backing-up-manually_{context}"]
= Backing up {product-title} data manually

You can back up {product-title} data manually at any time. Data backups are advised prior to any system updates. Backups are created in the `/var/lib/microshift-backups` directory by default. You can also specify a directory by adding custom parameters to the required backup command.

[NOTE]
====

> {product-title} creates an automatic backup on every start. RPM-OSTree systems backup and restore data automatically.

Reviewer: Same thing as other comment - there's no backup automation on regular, non-ostree systems

====

.Prerequisites
* You have root access to the host.

> * {product-title} and is stopped.

Reviewer: ```suggestion
* {product-title} is stopped.
```

.Procedure
. Manually create a backup using the default name and parent directory, `/var/lib/microshift-backups/<default-backup-name>`, by running the following command:
+
[source, terminal]
----
$ sudo microshift backup
----

. Optional. Manually create a backup with a specific name in the default directory by running the following command:
+
[source, terminal]
----

> $ sudo microshift backup --name /var/lib/microshift-backups/<my-custom-backup>/

Reviewer: ```suggestion
$ sudo microshift backup --name <my-custom-backup>
```

That `/var/lib/microshift-backups/` path can be changed using `--storage`, e.g.:
`--storage /mnt/another-backup-location`.

Reviewer 4: Can a documentation note be made that --name should not include the path? It causes the microshift backup command to produce an error

> $ sudo microshift backup --name <my-custom-backup>/

Reviewer: ```suggestion
$ sudo microshift backup --name <my-custom-backup>
```

----

. Optional. Manually create a backup in a specific parent directory by running the following command:
+
[source, terminal]
----

> $ sudo microshift backup --storage /var/lib/<custom-storage-location>

Reviewer: Technically, it doesn't strictly needs to be in the `/var/lib/` (though the gain of Copy-on-Write could be lost)

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-backing-up-manually_{context}"]
= Backing up {product-title} data manually

> You can back up {product-title} data manually at any time. Data backups are advised prior to any system updates. Backups are created in the `/var/lib/microshift-backups` directory by default. You can also specify a directory by adding custom parameters to the required backup command.

Reviewer 5: ```suggestion
You can back up {product-title} data manually at any time. Data backups are advised before any system updates. Backups are created in the `/var/lib/microshift-backups` directory by default. You can also specify a directory by adding custom parameters to the required backup command.
```
IBM Style lists "prior to" as a do-not-use term, in favor of "before".

[NOTE]
====
On `rpm-ostree` systems, {product-title} creates an automatic backup on every start and restores data automatically.
====

.Prerequisites
* You have root access to the host.
* {product-title} is stopped.

.Procedure

> . Manually create a backup using the default name and parent directory, `/var/lib/microshift-backups/<default-backup-name>`, by running the following command:

Reviewer 5: ```suggestion
. Manually create a backup that uses the default name and parent directory, `/var/lib/microshift-backups/<default-backup-name>`, by running the following command:
```
For clarity, replace "using" with either "that uses" or "by using", depending on your intended meaning.

??? I0829 07:32:12.318897    6586 version.go:227] "START reading version file"
??? I0829 07:32:12.319122    6586 version.go:233] "END reading version file" contents={"version":"4.14.0","deployment_id":"rhel-35d7b5c80f0f1378d6846f6dc1304bbf1dcdc5847198fcd4e6099364eaf99048.0","boot_id":"80364fcf3df54284a6902687e2cdd4c2"}
??? I0829 07:32:12.319256    6586 data_manager.go:82] "Copying data to backup directory" storage="/var/lib/microshift-backups" name="4.14.0__20230829_113212" data="/var/lib/microshift"
??? I0829 07:32:12.319423    6586 data_manager.go:186] "Starting copy" cmd="/bin/cp --verbose --recursive --preserve --reflink=auto /var/lib/microshift /var/lib/microshift-backups/4.14.0__20230829_113212"
??? I0829 07:32:12.324955    6586 data_manager.go:200] "Finished copy" cmd="/bin/cp --verbose --recursive --preserve --reflink=auto /var/lib/microshift /var/lib/microshift-backups/4.14.0__20230829_113212"
??? I0829 07:32:12.325014    6586 data_manager.go:115] "Copied data to backup directory" backup="/var/lib/microshift-backups/4.14.0__20230829_113212" data="/var/lib/microshift"
----

> . Optional. Manually create a backup with a specific name in the default directory by running the following command:

Reviewer 5: ```suggestion
. Optional: Manually create a backup with a specific name in the default directory by running the following command:
```
Per [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps), I believe we use a colon after the word _Optional_ in an optional step.

+
[source, terminal]
----
$ sudo microshift backup --name <my-custom-backup>
----

> . Optional. Manually create a backup in a specific parent directory by running the following command:

Reviewer 5: ```suggestion
. Optional: Manually create a backup in a specific parent directory by running the following command:
```
Per [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps), I believe we use a colon after the word _Optional_ in an optional step.

+
[source, terminal]
----
$ sudo microshift backup --storage /var/lib/<custom-storage-location>
----

> . Optional. Manually create a backup in a specific parent directory with a custom name by running the following command:

Reviewer 5: ```suggestion
. Optional: Manually create a backup in a specific parent directory with a custom name by running the following command:
```
Per [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps), I believe we use a colon after the word _Optional_ in an optional step.

+
[source, terminal]
----
$ sudo microshift backup --storage /var/lib/<custom-storage-location>/ --name <my-custom-backup>
----

.Verification

> . You can verify that the backup exists by viewing the data in the directory you chose. For example, `/var/lib/microshift-backups/<my-custom-backup>/` or `/var/lib/<custom-storage-location>/<my-custom-backup>/`.

Reviewer 5: ```suggestion
* You can verify that the backup exists by viewing the data in the directory you chose. For example, `/var/lib/microshift-backups/<my-custom-backup>/` or `/var/lib/<custom-storage-location>/<my-custom-backup>/`.
```
From the [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#single-step-procedures): "When a procedure contains only one step, use an unnumbered bullet."
